diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6203792..cffc563 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,10 +9,10 @@ endif()
 
 # ================================ General configuration ======================================
 
-project(PySide6QtAds)
+project(PySide2QtAds)
 
 # Find Python
-find_package (Python3 COMPONENTS Interpreter)
+find_package (Python3 COMPONENTS Interpreter Development)
 if(NOT Python3_Interpreter_FOUND)
     message(FATAL_ERROR "Python3 not found")
 endif()
@@ -22,19 +22,19 @@ message(STATUS "Using python: ${Python3_EXECUTABLE}")
 set(OpenGL_GL_PREFERENCE "LEGACY")
 
 # Find the required Qt packages
-find_package(Qt6 REQUIRED COMPONENTS Core Gui Widgets)
+find_package(Qt5 REQUIRED COMPONENTS Core Gui Widgets)
+find_package(Shiboken2 2.0.0 REQUIRED)
+find_package(PySide2 2.0.0 REQUIRED)
+find_package(qt5advanceddocking)
 
 # Set CPP standard to C++11 minimum.
 set(CMAKE_CXX_STANDARD 11)
 
 # The C++ project library for which we will create bindings.
-set(qtads_subdir "Qt-Advanced-Docking-System")
-set(qtads_dir ${CMAKE_CURRENT_SOURCE_DIR}/${qtads_subdir})
-add_subdirectory(${qtads_subdir} EXCLUDE_FROM_ALL)
-set(project_library "qtadvanceddocking")
+set(project_library "qt5advanceddocking")
 
 # The name of the generated bindings module (as imported in Python).
-set(bindings_library "PySide6QtAds")
+set(bindings_library "PySide2QtAds")
 
 # The header file with all the types and functions for which bindings will be generated.
 # Usually it simply includes other headers of the library you are creating bindings for.
@@ -57,65 +57,34 @@ set(generated_sources
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockcomponentsfactory_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockcontainerwidget_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockfocuscontroller_wrapper.cpp
-    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockingstatereader_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockmanager_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockoverlay_wrapper.cpp
-    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockoverlaycross_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdocksplitter_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockwidget_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cdockwidgettab_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_celidinglabel_wrapper.cpp
    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cfloatingdockcontainer_wrapper.cpp
-     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cfloatingdragpreview_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_ciconprovider_wrapper.cpp
-    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cspacerwidget_wrapper.cpp
-    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_ctitlebarbutton_wrapper.cpp
-    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_ifloatingwidget_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cpushbutton_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_cresizehandle_wrapper.cpp
     ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/ads_wrapper.cpp
-    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/pyside6qtads_module_wrapper.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/${bindings_library}/pyside2qtads_module_wrapper.cpp
     )
 
 
 # ================================== Shiboken detection ======================================
 
 
-# Macro to get various pyside / python include / link flags and paths.
-# Uses the not entirely supported utils/pyside_config.py file.
-macro(pyside_config option output_var)
-    if(${ARGC} GREATER 2)
-        set(is_list ${ARGV2})
-    else()
-        set(is_list "")
-    endif()
-
-    execute_process(
-      COMMAND ${Python3_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/pyside_config.py"
-              ${option}
-      OUTPUT_VARIABLE ${output_var}
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-    if ("${${output_var}}" STREQUAL "")
-        message(FATAL_ERROR "Error: Calling ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/pyside_config.py ${option} returned no output.")
-    endif()
-    if(is_list)
-        string (REPLACE " " ";" ${output_var} "${${output_var}}")
-    endif()
-endmacro()
-
 # Query for the shiboken generator path, Python path, include paths and linker flags.
-pyside_config(--shiboken-module-path shiboken_module_path)
-pyside_config(--shiboken-generator-path shiboken_generator_path)
-pyside_config(--pyside-path pyside_path)
-pyside_config(--pyside-include-path pyside_include_dir 1)
-pyside_config(--python-include-path python_include_dir)
-pyside_config(--shiboken-generator-include-path shiboken_include_dir 1)
-pyside_config(--shiboken-module-shared-libraries-cmake shiboken_shared_libraries 0)
-pyside_config(--python-link-flags-cmake python_linking_data 0)
-pyside_config(--pyside-shared-libraries-cmake pyside_shared_libraries 0)
-
-set(shiboken_path "${shiboken_generator_path}/shiboken6${CMAKE_EXECUTABLE_SUFFIX}")
+get_property(shiboken_include_dir TARGET Shiboken2::libshiboken PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(shiboken_shared_libraries TARGET Shiboken2::libshiboken PROPERTY IMPORTED_IMPLIB_RELEASE)
+get_property(pyside_include_dir TARGET PySide2::pyside2 PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(pyside_shared_libraries TARGET PySide2::pyside2 PROPERTY IMPORTED_IMPLIB_RELEASE)
+get_property(ads_include_dir TARGET ads::qt5advanceddocking PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(ads_link_libraries TARGET ads::qt5advanceddocking PROPERTY INTERFACE_LINK_LIBRARIES)
+get_property(ads_shared_libraries TARGET ads::qt5advanceddocking PROPERTY IMPORTED_IMPLIB_RELEASE)
+
+find_program(shiboken_path "shiboken2${CMAKE_EXECUTABLE_SUFFIX}")
 if(NOT EXISTS ${shiboken_path})
     message(FATAL_ERROR "Shiboken executable not found at path: ${shiboken_path}")
 endif()
@@ -123,14 +92,14 @@ endif()
 
 # ==================================== RPATH configuration ====================================
 
-set(CMAKE_SKIP_BUILD_RPATH TRUE)
+# set(CMAKE_SKIP_BUILD_RPATH TRUE)
 
 if(UNIX AND NOT APPLE)
-    set(CMAKE_INSTALL_RPATH "$ORIGIN/lib:$ORIGIN/../PySide6/:$ORIGIN/../PySide6/Qt/lib:$ORIGIN/../shiboken6")
+    set(CMAKE_INSTALL_RPATH "$ORIGIN/lib:$ORIGIN/../PySide2/:$ORIGIN/../PySide2/Qt/lib:$ORIGIN/../shiboken2")
 endif()
 
 if(APPLE)
-    set(CMAKE_INSTALL_RPATH "@loader_path/lib;@loader_path/../PySide6/;@loader_path/../PySide6/Qt/lib;@loader_path/../shiboken6")
+    set(CMAKE_INSTALL_RPATH "@loader_path/lib;@loader_path/../PySide2/;@loader_path/../PySide2/Qt/lib;@loader_path/../shiboken2")
     set(MACOSX_RPATH TRUE)
 endif()
 
@@ -138,7 +107,7 @@ endif()
 
 # Get the relevant Qt include dirs, to pass them on to shiboken.
 set(QT_INCLUDE_DIR "")
-get_target_property(QT_INCLUDE_DIR_LIST Qt6::Core INTERFACE_INCLUDE_DIRECTORIES)
+get_target_property(QT_INCLUDE_DIR_LIST Qt5::Core INTERFACE_INCLUDE_DIRECTORIES)
 foreach(_Q ${QT_INCLUDE_DIR_LIST})
     if(NOT "${_Q}" MATCHES "QtCore$")
         set(QT_INCLUDE_DIR "${_Q}")
@@ -151,26 +120,26 @@ endif()
 set(QT_INCLUDES "")
 list(APPEND QT_INCLUDES "-I${QT_INCLUDE_DIR}")
 
-get_property(QT_CORE_INCLUDE_DIRS TARGET Qt6::Core PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(QT_CORE_INCLUDE_DIRS TARGET Qt5::Core PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
 foreach(INCLUDE_DIR ${QT_CORE_INCLUDE_DIRS})
     list(APPEND QT_INCLUDES "-I${INCLUDE_DIR}")
 endforeach()
 
-get_property(QT_GUI_INCLUDE_DIRS TARGET Qt6::Gui PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(QT_GUI_INCLUDE_DIRS TARGET Qt5::Gui PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
 foreach(INCLUDE_DIR ${QT_GUI_INCLUDE_DIRS})
     list(APPEND QT_INCLUDES "-I${INCLUDE_DIR}")
 endforeach()
 
-get_property(QT_WIDGETS_INCLUDE_DIRS TARGET Qt6::Widgets PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(QT_WIDGETS_INCLUDE_DIRS TARGET Qt5::Widgets PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
 foreach(INCLUDE_DIR ${QT_WIDGETS_INCLUDE_DIRS})
     list(APPEND QT_INCLUDES "-I${INCLUDE_DIR}")
 endforeach()
 
 # Check if Qt is a framework build on macOS. This affects how include paths should be handled.
-get_target_property(QtCore_is_framework Qt6::Core FRAMEWORK)
+get_target_property(QtCore_is_framework Qt5::Core FRAMEWORK)
 if (QtCore_is_framework)
     # Get the path to the Qt framework dir.
-    set(QT_FRAMEWORK_INCLUDE_DIR "${QT6_INSTALL_PREFIX}/${QT6_INSTALL_LIBS}")
+    set(QT_FRAMEWORK_INCLUDE_DIR "${QT5_INSTALL_PREFIX}/${QT5_INSTALL_LIBS}")
     message(STATUS "*** QT_FRAMEWORK_INCLUDE_DIR is ${QT_FRAMEWORK_INCLUDE_DIR}")
     list(APPEND QT_INCLUDES "--framework-include-paths=${QT_FRAMEWORK_INCLUDE_DIR}")
 endif()
@@ -191,15 +160,20 @@ foreach(_current ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES})
     set(implicit_includes ${implicit_includes} "-I${_current}")
 endforeach()
 
+set(ads_includes)
+foreach(_current ${ads_include_dir})
+    set(ads_includes ${ads_includes} "-I${_current}")
+endforeach()
+
 # Set up the options to pass to shiboken.
 set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
     --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero
     --avoid-protected-hack
     ${QT_INCLUDES}
     ${implicit_includes}
-    -I${qtads_dir}/src
+    ${ads_includes}
     -T${CMAKE_CURRENT_SOURCE_DIR}
-    -T${pyside_path}/typesystems
+    -T${PYSIDE_TYPESYSTEMS}
     --output-directory=${CMAKE_CURRENT_BINARY_DIR}
     )
 
@@ -226,17 +200,21 @@ add_library(${bindings_library} MODULE ${${bindings_library}_sources})
 # Apply relevant include and link flags.
 target_include_directories(${bindings_library} PRIVATE ${pyside_additional_includes})
 target_include_directories(${bindings_library} PRIVATE ${pyside_include_dir})
-target_include_directories(${bindings_library} PRIVATE ${python_include_dir})
+target_include_directories(${bindings_library} PRIVATE ${Python3_INCLUDE_DIRS})
 target_include_directories(${bindings_library} PRIVATE ${shiboken_include_dir})
-target_include_directories(${bindings_library} PRIVATE "${CMAKE_SOURCE_DIR}/src")
+target_include_directories(${bindings_library} PRIVATE ${ads_include_dir})
 
-target_link_libraries(${bindings_library} PRIVATE Qt6::Widgets)
-target_link_libraries(${bindings_library} PRIVATE ${project_library})
+target_link_libraries(${bindings_library} PRIVATE Qt5::Widgets)
 target_link_libraries(${bindings_library} PRIVATE ${pyside_shared_libraries})
 target_link_libraries(${bindings_library} PRIVATE ${shiboken_shared_libraries})
+target_link_libraries(${bindings_library} PRIVATE ${ads_link_libraries})
+target_link_libraries(${bindings_library} PRIVATE ${ads_shared_libraries})
+target_link_libraries(${bindings_library} PUBLIC ads::qt5advanceddocking)
+
+if (UNIX)
+    target_link_libraries(${bindings_library} PUBLIC PySide2::pyside2)
+endif()
 
-target_compile_options(qtadvanceddocking PRIVATE -Og)
-target_compile_options(${bindings_library} PRIVATE -Og)
 target_compile_definitions(${bindings_library} PRIVATE "-DPy_LIMITED_API=0x03070000")
 
 # Adjust the name of generated module.
@@ -253,11 +231,9 @@ if(APPLE)
 endif(APPLE)
 
 if (WIN32)
-    list(GET python_linking_data 0 python_libdir)
-    target_link_directories(${bindings_library} PRIVATE ${python_libdir})
+    target_link_directories(${bindings_library} PRIVATE ${Python3_LIBRARY_DIRS})
 endif()
 
-
 # ================================= Dubious deployment section ================================
 
 install(TARGETS ${bindings_library}
@@ -265,16 +241,11 @@ install(TARGETS ${bindings_library}
         RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}
         )
 
-if(NOT BUILD_STATIC)
-    install(TARGETS ${project_library}
-            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            )
-endif()
+install(FILES ${hinting_stub_files}
+        DESTINATION "${CMAKE_INSTALL_PREFIX}")
 
 install(FILES
-    "${qtads_dir}/LICENSE"
-    "${qtads_dir}/gnu-lgpl-v2.1.md"
+    "LICENSE"
     DESTINATION license/ads
     COMPONENT license
 )
